File: kunal/2966. Divide Array Into Arrays With Max Difference.py
class Solution:
    def divideArray(self, nums: list[int], k: int) -> list[list[int]]:
        subarray_count = len(nums) // 3
        ans = [[] for _ in range(subarray_count)]

        nums.sort()

        for i in range(0, len(nums), +3):
            if (nums[i + 2] - nums[i]) <= k:
                ans[i // 3].extend([nums[i], nums[i + 1], nums[i + 2]])
            else:
                return []

        return ans


print(Solution().divideArray([1, 3, 4, 8, 7, 9, 3, 5, 1], 2))
print(Solution().divideArray([1, 3, 3, 2, 7, 3], 3))

File: kunal/2966. Divide Array Into Arrays With Max Difference.java
import java.util.Arrays;

class Solution {
    public int[][] divideArray(int[] nums, int k) {
        int subarray_count = nums.length / 3;
        int[][] ans = new int[subarray_count][3];

        Arrays.sort(nums);

        for (int i = 0; i < nums.length; i = i + 3) {
            if ((nums[i + 2] - nums[i]) <= k) {
                ans[i / 3][0] = nums[i];
                ans[i / 3][1] = nums[i + 1];
                ans[i / 3][2] = nums[i + 2];
            } else {
                return new int[0][0];
            }
        }

        return ans;
    }
}
